utils/synth_wav.sh

```diff
@@ -0,0 +1,250 @@
+fi
+if [ -z "${synth_model}" ]; then
+    download_models
+    synth_model=$(find ${download_dir}/${models} -name "model*.best" | head -n 1)
+fi
+if [ -z "${decode_config}" ]; then
+    download_models
+    decode_config=$(find ${download_dir}/${models} -name "decode*.yaml" | head -n 1)
+fi
+if [ -z "${txt}" ]; then
+    download_models
+    txt=$(find ${download_dir}/${models}/etc -name "*.txt" | head -n 1)
+fi
+
+# Check file existence
+if [ ! -f "${cmvn}" ]; then
+    echo "No such CMVN file: ${cmvn}"
+    exit 1
+fi
+if [ ! -f "${dict}" ]; then
+    echo "No such dictionary: ${dict}"
+    exit 1
+fi
+if [ ! -f "${input_wav}" ] && [ ${use_input_wav} ]; then
+    echo "No such WAV file for extracting meta information: ${input_wav}"
+    exit 1
+fi
+if [ ! -f "${synth_model}" ]; then
+    echo "No such E2E model: ${synth_model}"
+    exit 1
+fi
+if [ ! -f "${decode_config}" ]; then
+    echo "No such config file: ${decode_config}"
+    exit 1
+fi
+if [ ! -f "${txt}" ]; then
+    echo "No such txt file: ${txt}"
+    exit 1
+fi
+
+base=$(basename $txt .txt)
+decode_dir=${decode_dir}/${base}
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    echo "stage 0: Data preparation"
+
+    mkdir -p ${decode_dir}/data
+    echo "$base ${decode_dir}/data/dummy.wav" > ${decode_dir}/data/wav.scp
+    echo "X $base" > ${decode_dir}/data/spk2utt
+    echo "$base X" > ${decode_dir}/data/utt2spk
+    echo -n "$base " > ${decode_dir}/data/text
+    cat $txt >> ${decode_dir}/data/text
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    echo "stage 1: Dummy Feature Generation"
+
+    sox -n -r ${fs} -c 1 -b 16 ${decode_dir}/data/dummy.wav trim 0.000 0.001
+    make_fbank.sh --cmd "${train_cmd}" --nj 1 \
+        --fs ${fs} \
+        --fmax "${fmax}" \
+        --fmin "${fmin}" \
+        --n_fft ${n_fft} \
+        --n_shift ${n_shift} \
+        --win_length "${win_length}" \
+        --n_mels ${n_mels} \
+        ${decode_dir}/data \
+        ${decode_dir}/log \
+        ${decode_dir}/fbank
+
+    feat_synth_dir=${decode_dir}/dump; mkdir -p ${feat_synth_dir}
+    dump.sh --cmd "$train_cmd" --nj 1 --do_delta false \
+        ${decode_dir}/data/feats.scp ${cmvn} ${decode_dir}/log ${feat_synth_dir}
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    echo "stage 2: Json Data Preparation"
+
+    feat_synth_dir=${decode_dir}/dump
+    data2json.sh --feat ${feat_synth_dir}/feats.scp \
+        ${decode_dir}/data ${dict} > ${feat_synth_dir}/data.json
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && [ ${use_input_wav} ]; then
+    echo "stage 3: x-vector extraction"
+
+    utils/copy_data_dir.sh ${decode_dir}/data ${decode_dir}/data2
+    echo "$base ${input_wav}" > ${decode_dir}/data2/wav.scp
+    utils/data/resample_data_dir.sh 16000 ${decode_dir}/data2
+    steps/make_mfcc.sh \
+        --write-utt2num-frames true \
+        --mfcc-config conf/mfcc.conf \
```

Comment:
Maybe `mfcc.conf` is missing?

Reply:
OK, I will change the directory.

Reply:
What do you think about that we include `mfcc.conf` and `vad.conf` in a tar.gz file?

Reply:
I think current style is acceptable.

Reply:
OK.